_layouts/release.html

```diff
@@ -5,7 +5,11 @@
 <article class="release release">
     <header class="release_header">
         <h1 class="release_title">{{ page.title }}</h1>
+        {% if page.release_tag == nil %}
         <p class="release_meta"><time datetime="{{ page.date | date_to_xmlschema }}" itemprop="datePublished">{{ page.date | date: "%A, %Y-%m-%d" }}</time></p>
+        {% else %}
+        <p class="release_meta">Release: {{ page.release_tag }}</p>
+        {% endif %}
```

Comment:
I prefer to avoid keeping those ifs for "backwards compatibility".
I'd modify the old release files, adding the release_tag. If we don't want or don't have tags for older commits, we could simply write the date in the release_tag field, so the page will still be generated with the same content.